src/app/groups/containers/group-header/group-header.component.ts

```diff
@@ -1,16 +1,43 @@
-import { Component, Input } from '@angular/core';
+import { Component, input } from '@angular/core';
 import { GroupData } from '../../services/group-datasource.service';
-import { NgIf } from '@angular/common';
+import { AsyncPipe, JsonPipe, NgIf } from '@angular/common';
+import { GetItemByIdService } from 'src/app/data-access/get-item-by-id.service';
+import { distinctUntilChanged, map, switchMap } from 'rxjs/operators';
+import { of, merge, Subject } from 'rxjs';
+import { toObservable } from '@angular/core/rxjs-interop';
+import { fetchingState } from 'src/app/utils/state';
+import { LoadingComponent } from 'src/app/ui-components/loading/loading.component';
+import { ErrorComponent } from 'src/app/ui-components/error/error.component';
+import { mapToFetchState } from 'src/app/utils/operators/state';
+import { ButtonDirective } from 'primeng/button';
 
 @Component({
   selector: 'alg-group-header',
   templateUrl: './group-header.component.html',
   styleUrls: [ './group-header.component.scss' ],
   standalone: true,
-  imports: [ NgIf ],
+  imports: [ NgIf, AsyncPipe, JsonPipe, LoadingComponent, ErrorComponent, ButtonDirective ],
 })
 export class GroupHeaderComponent {
-  @Input() groupData?: GroupData;
+  groupData = input.required<GroupData>();
 
-  constructor() {}
+  private refreshSubject = new Subject<void>();
+  state$ = toObservable(this.groupData).pipe(
+    map(groupData => groupData.group.rootActivityId),
+    distinctUntilChanged(),
+    switchMap(rootActivityId =>
+      (rootActivityId === null ? of(null) : merge(
+        of(fetchingState()),
```

Comment:
Fixed,

The problem was with `isFetching` value when activity changed (didn't change it to true). It's because in first my draft the stream was flat